src/app.tsx

```diff
@@ -0,0 +1,438 @@
+        alt={`${props.bugType}`}
+        style={isOnHover ? hoverBugImgStyle : bugImgStyle}
+        onMouseOver={changeHoverStyle}
+        onMouseLeave={changeHoverStyle}
+      />
+    </foreignObject>
+  );
+};
+
+interface BugItem {
+  bugType: string;
+  imgSrc: string;
+  isFriendly: boolean;
+}
+
+// this enum stores all values associated with objects in the game
+// also helps prevent naming inconsistencies
+
+const BUGS: { [key: string]: BugItem } = {
+  LADY_BUG: {
+    bugType: "LADY_BUG",
+    imgSrc: ladyBugImg,
+    isFriendly: true,
+  },
+  BEE: {
+    bugType: "BEE",
+    imgSrc: beeImg,
+    isFriendly: true,
+  },
+  MITE: {
+    bugType: "MITE",
+    imgSrc: miteImg,
+    isFriendly: false,
+  },
+  SLUG: {
+    bugType: "SLUG",
+    imgSrc: slugImg,
+    isFriendly: false,
+  },
+};
+
+const bugTypes = Object.keys(BUGS);
+
+interface ProgressBarProps {
+  barLength: number;
+  progressPercent: number;
+}
+
+const ProgressBar: React.FC<ProgressBarProps> = (props) => {
+  const [progressBarLength, setProgresBarLength] = useState(0);
+
+  // runs every time the component rerenders
+  useEffect(() => {
+    setProgresBarLength(props.progressPercent * props.barLength);
+  }, [props.progressPercent]);
+
+  // const progressBarContainerStyle = {
+  //   Position: "absolute",
+  //   left: "100px",
+  //   top: "100px",
+  //   display: "flex",
+  //   alignItems: "center",
+  //   justifyContent: "center",
+  //   ZIndex: "10",
+  // };
+
+  const progressBarStyle = {
+    backgroundColor: "rgb(233, 233, 233)",
+    borderRadius: "0.5rem",
+    width: `${props.barLength}px`,
+  };
+
+  const filledProgressBarStyle = {
+    backgroundColor: "rgb(62, 122, 235)",
+    height: "10px",
+    borderRadius: "1rem",
+    width: `${progressBarLength}px`,
+    // animation
+    transition: "1s ease",
+  };
+
+  return (
+    <div className="absolute top-16 left-16 z-10">
+      <div style={progressBarStyle}>
+        <div style={filledProgressBarStyle} />
+      </div>
+    </div>
+  );
+};
+
+interface GameProps {}
+
+// main parent component
+const Game: React.FC<GameProps> = (props) => {
+  // define states using state hook
+  // typescript implicitly knows the type of each state from initial value
+  const [showModal, setShowModal] = useState(false);
+  const [currentBugType, setCurrentBugType] = useState(""); // so that modal component knows a bug is good or bad
+  const [hasWon, setHasWon] = useState(false);
+  const [correctBugs, setCorrectBugs] = useState(new Set()); // tracking progress
+  const [bugIds, setBugIds] = useState<number[]>([]); // an array of bug ids
+
+  const intervalID = useRef(0);
+
+  // this variable keeps track of total # of spawned bugs
+  // also sets the id for a newest spawned bug
+  const numSpawnedBugs = useRef(0);
+
+  // component did mount
+  useEffect(() => {
+    // create a new bug and give it a unique index
+
+    const spawnNewBug = () => {
+      // if (n > UPPER_COUNT) return
```

Comment:
Redundant comment. Comments should explain "why" rather than "what" the code does.